src/WebJobs.Script/DependencyInjection/ScriptStartupTypeLocator.cs

```diff
@@ -102,10 +102,9 @@ public async Task<IEnumerable<Type>> GetExtensionsStartupTypesAsync()
                     }
                     isPrecompiledFunctionApp = isPrecompiledFunctionApp || functionMetadata.Language == DotNetScriptTypes.DotNetAssembly;
                 }
-
-                isDotnetIsolatedApp = IsDotnetIsolatedApp(functionMetadataCollection, SystemEnvironment.Instance);
             }
 
+            bool isDotnetIsolatedApp = IsDotnetIsolatedApp(functionMetadataCollection, SystemEnvironment.Instance);
```

Comment:
The below test was failing on [this](https://github.com/Azure/azure-functions-host/blob/266f0fb78575bfe3b8182c21cf20fd01a64f794b/test/WebJobs.Script.Tests/ScriptStartupTypeDiscovererTests.cs#L512) line periodically. This was because IsDotnetIsolatedApp wasn't correct sometimes, causing changes to the bundles path resolution and loading?

Are you sure this is only a logging issue? The code below seems to resolve the path differently based on this value. Could apps be using the wrong path?